scripts/support/create_ge_images/create_template.sh

```diff
@@ -0,0 +1,193 @@
+  echo "VM id is $ID"
+  sleep 5
+  nova floating-ip-associate $ID $IP
+  # Wait until ssh is ready
+  ussh="ssh -oStrictHostKeyChecking=no"
+  ssh-keygen -f ~/.ssh/known_hosts -R $IP
+  echo "Waiting until ssh is ready..."
+  sleep 30
+  until $ussh ${user}@${IP} true 2>/dev/null ; do sleep 10 ; done
+
+  echo "VM is ready"
+
+  # if present, upload file
+  if [ -n "${UPLOAD_FILE}" -a -f scripts/$UPLOAD_FILE ] ; then
+     echo "Uploading scripts/${UPLOAD_FILE}"
+     scp scripts/$UPLOAD_FILE ${user}@${IP}:
+  fi
+
+  # upload create script
+  scp scripts/$CREATESCRIPT ${user}@${IP}:create.sh
+  ssh ${user}@${IP} chmod 700 create.sh 
+
+  # run script, delete support account,  poweroff the VM
+  # Centos requires -t -t because sudo needs tty; with Debian -t -t sometimes 
+  # hungs with apt-get upgrade.
+  if [ "$user" = "centos" ] ; then
+    echo 'Running script in CentOS distribution'
+    O="-t"
+  else
+    echo 'Running script in Debian/Ubuntu distribution'
+    O=""
+  fi
+  set -o pipefail
+  ssh $O ${user}@${IP} ./create.sh 2>&1 |tee create.log
+  set +o pipefail
+  ssh $O ${user}@${IP} sudo userdel support
+  ssh $O ${user}@${IP} sudo rm -rf /home/support create.sh $UPLOAD_FILE
+  ssh $O ${user}@${IP} sudo poweroff || true
+
+  # create template from VM, delete VM
+  sleep 20
+  nova image-create ${ID} ${name}-snapshot --poll
+  nova delete ${ID}
+
+  # download snapshot and delete it from server
+  echo "downloading image"
+  glance image-download --file tmp_image ${name}-snapshot
+  echo "deleting image"
+  glance image-delete ${name}-snapshot
+
+  # sysprep VM
+  echo "running virt-sysprep"
+  sudo virt-sysprep -a tmp_image
+
+  # shrink VM
+  sudo ./shrink.py
+  echo "running virt-sparsify"
+  sudo virt-sparsify  tmp_image tmp_image.new
+  rm -f tmp_image
+  mv tmp_image.new tmp_image
+
+  # boot centOS images and poweroff after reboot; this is required because
+  # SELinux relabeling.
+  if [ "$user" = "centos" ] ; then
+      sudo kvm  -no-reboot -nographic  -m 2048 -hda tmp_image
+  fi
+
+  # upload template and delete file
+  echo "uploading image"
+  glance image-create --name ${name}_rc --disk-format qcow2 --container-format bare --is-public False --file tmp_image
+  rm -f tmp_image
+
+  # test template
+  test_template $name $user
+}
+
+test_template() {
+  name=$1
+  user=$2
+
+  # launch a VM using the new template
+  echo "Launching a testing VM"
+  ID=$(nova boot test-${name} --security-groups $SECURITY_GROUP_TEST --flavor m1.small --key-name $KEYPAIR --image ${name}_rc --nic net-id=$SHARED_NETWORK_ID | awk -F\| '/\|[ \t]+id[ \t]+\|/ {print $3}')
+  echo "VM id: $ID"
+  sleep 5
+  nova floating-ip-associate $ID $IP
+  # Wait until ssh is reading
+  ussh="ssh -oStrictHostKeyChecking=no"
+  ssh-keygen -f ~/.ssh/known_hosts -R $IP
+  echo "Waiting until ssh is ready..."
+  sleep 30
+  until $ussh ${user}@${IP} true 2>/dev/null ; do sleep 10 ; done
```

Comment:
Idem to previous one, but maybe if you are making the same operation you could create a function to do it.